osfoffline/views/start_screen.py

```diff
@@ -21,47 +32,67 @@ def __init__(self):
-            for user in session.query(User):
-                if user.osf_login == user_name:
+            resp = yield from osf.make_request(url, expects=[200])
+        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, aiohttp.errors.TimeoutError):
+            # No internet connection
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Unable to connect to server. Check your internet connection or try again later."
+            )
+        except aiohttp.errors.HttpBadRequest:
+            # Invalid credentials
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Invalid login credentials."
+            )
+        else:
+            json_resp = yield from resp.json()
+            remote_user = RemoteUser(json_resp['data'])
+            full_name = json_resp['data']['attributes']['full_name']     # Use one of these later to "Welcome {user}"
+            given_name = json_resp['data']['attributes']['given_name']   # Use one of these later to "Welcome {user}"
+            osf_id = json_resp['data']['id']
+
+            try:
+                user = session.query(User).filter(User.osf_id == remote_user.id).one()
+            except MultipleResultsFound:
+                logging.warning('multiple users with same username. deleting all users with this username. restarting function.')
+                for user in session.query(User).filter(User.osf_id == remote_user.id).all():
                     session.delete(user)
                     save(session)
-            self.log_in()
-        except NoResultFound:
-            logging.warning('user doesnt exist. Creating user. and logging them in.')
-            user = User(
-                full_name=full_name,
-                osf_id=osf_id,
-                osf_login=user_name,
-                osf_local_folder_path='',
-                oauth_token=oauth_token,
-                osf_password=password
-            )
-            user.logged_in = True
-            save(session, user)
+                self.log_in()
+            except NoResultFound:
+                logging.debug('user doesnt exist. Creating user. and logging them in.')
+                user = User(
+                    full_name=full_name,
+                    osf_id=osf_id,
+                    osf_login='',  # TODO: email goes here when more auth methods are added, not currently returned by APIv2
+                    osf_local_folder_path='',
+                    oauth_token=personal_access_token,
+                )
+                user.logged_in = True
+                save(session, user)
+            else:
+                if not user.oauth_token == personal_access_token:
+                    user.oauth_token == personal_access_token
+                user.logged_in = True
+                save(session, user)
```

Comment:
The user.logged_in and save calls are still duplicated (see line 83); why not move outside the try-except altogether?